### DISCUSSION TITLE
Pan/zoom overlaid element with deck controller

### DISCUSSION THREAD

**Discussion by kwala007** [visgl/deck.gl#9101] [Q&A]
Hello, I have a few SVGs overlaid on a deck.gl map (pure JS implementation). I'd like to be able to move them around with the map when I zoom/pan using the built in MapController. I assume there is some sort of transformation applied to the map view every time a user manipulates the view. **Is there any way to extract that transformation from the controller's event handler so it can be applied to the overlaid SVGs as well?** I've managed to extract the relevant events themselves using a custom controller, but I need to know how the controller handles the events to ensure the map and SVGs move together. I can't put the SVGs into an IconLayer, since the SVGs are interactively added, moved, and deleted by the user. I am a JS beginner, so apologies if there is a simple solution I missed.

**Comment by TextTL** [comment]
There's one example here: https://github.com/visgl/deck.gl/issues/3945
